Sources/Console/Activity/CustomeActivityIndicator.swift

```diff
@@ -0,0 +1,115 @@
+    ///         indicator.succeed()
+    ///     }
+    ///     // start the indicator and wait for it to finish
+    ///     try indicator.start(on: ...).wait()
+    ///
+    /// - Note: If you want some ideas for indicator styles, take a look here:
+    ///   https://github.com/kiliankoe/CLISpinner/blob/master/Sources/CLISpinner/Pattern.swift#L88-L151
+    ///
+    /// - Parameters:
+    ///   - frames: The strings to loop over as the activity indicator runs.
+    ///   - success: The string to replace the indicator with when the operation succeeds. The default value is `[Done]`.
+    ///   - failure: The string to replace the indicator with when the operation fails: The default value is `[Failed]`.
+    ///   - color: The color of text when the frames are displayed. The default value is `.cyan`.
+    ///
+    /// - Returns: An `ActivityIndicator` that can start and stop the indicator.
+    public func activity(
+        frames: [String], success: String = "[Done]", failure: String = "[Failed]", color: ConsoleColor = .cyan
+    ) -> ActivityIndicator<CustomActivityIndicator> {
+        return CustomActivityIndicator(frames: frames, success: success, failure: failure, color: color).newActivity(for: self)
+    }
+    
+    /// Creates an activity indicator with custom frames that are iterated over.
+    ///
+    ///     // Create an activity indicator with the strings (frames) to loop over as it runs.
+    ///     let indicator = console.activity(frames: ["⠋","⠙","⠹","⠸","⠼","⠴","⠦","⠧","⠇","⠏"])
+    ///
+    ///     background {
+    ///         // complete the indicator after 3 seconds
+    ///         console.blockingWait(seconds: 3)
+    ///         indicator.succeed()
+    ///     }
+    ///     // start the indicator and wait for it to finish
+    ///     try indicator.start(on: ...).wait()
+    ///
+    /// - Note: If you want some ideas for indicator styles, take a look here:
+    ///   https://github.com/kiliankoe/CLISpinner/blob/master/Sources/CLISpinner/Pattern.swift#L88-L151
+    ///
+    /// - Parameters:
+    ///   - frames: The text to loop over as the activity indicator runs.
+    ///   - success: The string to replace the indicator with when the operation succeeds. The default value is `[Done]`.
+    ///   - failure: The string to replace the indicator with when the operation fails: The default value is `[Failed]`.
+    ///
+    /// - Returns: An `ActivityIndicator` that can start and stop the indicator.
+    public func activity(
+        frames: [ConsoleText], success: String = "[Done]", failure: String = "[Failed]"
+    ) -> ActivityIndicator<CustomActivityIndicator> {
+        return CustomActivityIndicator(frames: frames, success: success, failure: failure).newActivity(for: self)
+    }
+}
+
+/// An activity indicator with customizable frames and success and failure messages.
+///
+/// See `Console.activity(frames:success:failure:color:)` to make one.
+public struct CustomActivityIndicator: ActivityIndicatorType {
```

Comment:
This name is a bit confusing since there is a type called `ActivityIndicator` and this is not actually a custom version of it. This is a custom _ActivityIndicatorType_. I think it might be better to name this just `CustomActivity`.